src/lib/provable/merkle-tree-indexed.ts

```diff
@@ -701,3 +709,57 @@ function bisectUnique(
 
   return { lowIndex: iLow, foundValue: getValue(iLow) === target };
 }
+
+// custom comparison methods where 0 can act as the min and max value simultaneously
+
+/**
+ * Assert that `x in (low, high)`, i.e. low < x < high, with the following exceptions:
+ *
+ * - high=0 is treated as the maximum value, so `x in (low, 0)` always succeeds if only low < x; except for x = 0.
+ * - x=0 is also treated as the maximum value, so `0 in (low, high)` always fails, because x >= high.
+ */
+function assertStrictlyBetween(
```

Comment:
should we move that to somewhere close to the range check gadgets?

Reply:
I don't know, because it's so very specialized to this use case